test/run/test_assemblers.py

```diff
@@ -25,24 +25,23 @@ def setUp(self):
         self.osbuild = test.OSBuild(self)
 
     @contextlib.contextmanager
-    def run_assembler(self, name, options, output_path):
-        with self.osbuild as osb:
-            with open(os.path.join(self.locate_test_data(),
-                                   "manifests/filesystem.json")) as f:
-                manifest = json.load(f)
-            manifest["pipeline"] = dict(
-                manifest["pipeline"],
-                assembler={"name": name, "options": options}
-            )
-            data = json.dumps(manifest)
-
-            treeid = osb.treeid_from_manifest(data)
-            assert treeid
-
-            osb.compile(data, checkpoints=[treeid])
-            with osb.map_object(treeid) as tree, \
-                 osb.map_output(output_path) as output:
-                yield tree, output
+    def run_assembler(self, osb, name, options, output_path):
+        with open(os.path.join(self.locate_test_data(),
+                               "manifests/filesystem.json")) as f:
+            manifest = json.load(f)
+        manifest["pipeline"] = dict(
+            manifest["pipeline"],
+            assembler={"name": name, "options": options}
+        )
+        data = json.dumps(manifest)
+
+        treeid = osb.treeid_from_manifest(data)
+        assert treeid
+
+        osb.compile(data, checkpoints=[treeid])
```

Comment:
This checkpoints the assembled image. If you run this on a couple of images, you easily exceed the 16gig space we have on GHW. I know we don't do this right now, but I am afraid that we just run into space issues again and the only way out is to stop sharing the cache.

I would rather have us only put things in the cache that we actually want to share. However, I did not find a nice way to get access to intermediate pipeline artifacts via other means than checkpointing. Any ideas?

Reply:
I think that in the case of assembler tests, the issue is not in the checkpointing as the `treeid` is always the same for all pipelines run in one test case (we always, with the exception of `ostree`, build the `filesystem.json` manifest).

The issue here is that the resulting images are stored in the same directory each time. However, we can solve this:

1) I can make the resulting images way smaller (100 MB?) because we don't install much in them.
2) `osb.compile` can take the output directory as an argument, which can be temporary for just one test run. Doesn't sound too shabby to me...

Reply:
> I think that in the case of assembler tests, the issue is not in the checkpointing as the `treeid` is always the same for all pipelines run in one test case (we always, with the exception of `ostree`, build the `filesystem.json` manifest).

We manually patch the manifest, don't we? The assembler options are modified in `run_assembler()`, so the pipeline ID should be different on each run.

> The issue here is that the resulting images are stored in the same directory each time. However, we can solve this:
> 
>     1. I can make the resulting images way smaller (100 MB?) because we don't install much in them.

Yeah, I fully agree that we could use much simpler images there. I thought we already merged one improvement there last week?

>     2. `osb.compile` can take the output directory as an argument, which can be temporary for just one test run. Doesn't sound too shabby to me...

I am fine with sharing it, as long as we just delete the content after each run. Could even be a helper that clears it. However, note that the assembler tests always checkpoint into the cache as well, because they need the un-assembled tree as comparison.

Reply:
Ah, as Tom pointed out, the assembler patches only affect the assembler-ID, not the pipeline-ID before that. I missed that. So yeah, this should actually be fine.